### PR TITLE
include leading :colon in named params

### LINKED ISSUE #22: PDO bindParam should support named parameters with leading colon :
Consider the following statement:

        INSERT INTO `doc`.`tweets` (`id`, `text`, `user`) VALUES (:qp0, :qp1, :qp2)

When I prepare the statement, the value of namedToPositionalParams is:

        array(3) {
        ["qp0"]=>
        int(0)
        ["qp1"]=>
        int(1)
        ["qp2"]=>
        int(2)
        }

When I call:

        $pdoStatement->bindParam(":qp0", $value, $dataType);

My statement fails with:

        Exception\UnsupportedException("0-based parameter binding not supported, use 1-based");

This is because of line 277 of PDOStatement.php:

        if (array_key_exists($parameter, $this->namedToPositionalMap))

The keys in namedToPositionalParams should have leading colons, such as [":qp0"=>0, ":qp1"=>1, ":qp2"=>2].  http://php.net/manual/en/pdostatement.bindparam.php states: Parameter identifier. For a prepared statement using named placeholders, this will be a parameter name of the form :name

This can be fixed by moving the pattern's capturing group to include the colon on line 109 of PDOStatement.php:

        $pattern = '/(:(?:[\w|\d|_](?=([^\'\\\]*(\\\.|\'([^\'\\\]*\\\.)*[^\'\\\]*\'))*[^\']*$))*)/';

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/16059652-pdo-bindparam-should-support-named-parameters-with-leading-colon?utm_campaign=plugin&utm_content=tracker%2F5517944&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5517944&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by caleblloyd**
fixes #22

### CODE REVIEW

File: test/CrateIntegrationTest/PDO/PDOStatementTest.php [hunk 1]
```diff
@@ -202,8 +202,8 @@ public function testBindNamedParam()
         $name = 'second';
         $id = 2;
         $statement = $this->pdo->prepare('SELECT * FROM test_table where name = :name and id = :id');
-        $statement->bindParam('name', $name);
```
Comment (Ocramius) [inline test/CrateIntegrationTest/PDO/PDOStatementTest.php]: ***HUGE*** BC break?

Comment (caleblloyd) [inline test/CrateIntegrationTest/PDO/PDOStatementTest.php]: Would you prefer a solution that optionally accepts the leading :colon to maintain BC?
I was just trying to implement it to spec http://php.net/manual/en/pdostatement.bindparam.php

Comment (Ocramius) [inline test/CrateIntegrationTest/PDO/PDOStatementTest.php]: Yeah, BC is important, heh

**Comment by caleblloyd** [comment]
closing in favor of pr #24, which maintains BC

**Pull request closed without merging**